tests/core/chain-object/test_chain_reorganization.py

```diff
@@ -127,28 +79,42 @@ def test_import_block_with_reorg(chain, fork_chain, funded_address_private_key):
 
 def test_import_block_with_reorg_with_current_head_as_uncle(
         chain,
-        fork_chain,
         funded_address_private_key):
     """
     https://github.com/ethereum/py-evm/issues/1185
     """
-    # mine a block on the main chain which will eventually become an uncle on
-    # the main chain after a reorg.
-    block = chain.mine_block()
-
-    # Force the fork_chain to diverge from the main chain
-    fork_chain.header = fork_chain.header.copy(extra_data=b'fork-it!')
-    f_block_a = fork_chain.mine_block()
-
-    # now mine a block which has the current chain head as an uncle.
-    assert f_block_a != block
-    f_block_b = fork_chain.mine_block(uncles=(block.header,))
-
-    # ensure that we don't cause a re-org with our first import.
-    _, new_chain, _ = chain.import_block(f_block_a)
-    assert new_chain == tuple()
+    def converge_split_fn(results):
+        chain_a = results[0][-1]
+        chain_b = results[1][-1]
+
+        # the chain head from the main chain will become an uncle once we
+        # import the fork chain blocks.
+        uncle = chain_a.get_canonical_head()
+
+        pipe(
+            chain_b,
```

Comment:
more reliance on mutation in `pipe` :/

Reply:
I agree,  :+1: on a `build(chain, fn(1), fn(1))` function which is equivalent to `pipe(chain, copy(), fn1(), fn2()).